rust/lance-encoding/src/encodings/physical/fixed_size_binary.rs

```diff
@@ -0,0 +1,388 @@
+
+        let test_cases = TestCases::default()
+            .with_range(0..2)
+            .with_range(0..3)
+            .with_range(1..3)
+            .with_indices(vec![0, 1, 3, 4]);
+
+        check_round_trip_encoding_of_data(
+            vec![Arc::new(string_array)],
+            &test_cases,
+            HashMap::new(),
+        )
+        .await;
+    }
+
+    #[test_log::test(tokio::test)]
+    async fn test_fixed_size_sliced_utf8() {
+        let string_array = StringArray::from(vec![Some("abc"), Some("def"), None, Some("fgh")]);
+        let string_array = string_array.slice(1, 3);
+
+        let test_cases = TestCases::default()
+            .with_range(0..1)
+            .with_range(0..2)
+            .with_range(1..2);
+        check_round_trip_encoding_of_data(
+            vec![Arc::new(string_array)],
+            &test_cases,
+            HashMap::new(),
+        )
+        .await;
+    }
+
+    #[test_log::test(tokio::test)]
+    async fn test_bigger_than_max_page_size() {
+        // Create an array with one single 32MiB string
+        let big_string = String::from_iter((0..(32 * 1024 * 1024)).map(|_| '0'));
+        let string_array = StringArray::from(vec![
+            Some(big_string),
+            Some("abc".to_string()),
+            None,
+            None,
+            Some("xyz".to_string()),
+        ]);
+
+        // Drop the max page size to 1MiB
+        let test_cases = TestCases::default().with_max_page_size(1024 * 1024);
+
+        check_round_trip_encoding_of_data(
+            vec![Arc::new(string_array)],
+            &test_cases,
+            HashMap::new(),
+        )
+        .await;
+
+        // This is a regression testing the case where a page with X rows is split into Y parts
+        // where the number of parts is not evenly divisible by the number of rows.  In this
+        // case we are splitting 90 rows into 4 parts.
+        let big_string = String::from_iter((0..(1000 * 1000)).map(|_| '0'));
+        let string_array = StringArray::from_iter_values((0..90).map(|_| big_string.clone()));
+
+        check_round_trip_encoding_of_data(
+            vec![Arc::new(string_array)],
+            &TestCases::default(),
+            HashMap::new(),
+        )
+        .await;
+    }
+
+    #[test_log::test(tokio::test)]
+    async fn test_fixed_size_empty_strings() {
+        // All strings are empty
+
+        // When encoding an array of empty strings there are no bytes to encode
+        // which is strange and we want to ensure we handle it
+        let string_array = Arc::new(StringArray::from(vec![Some(""), None, Some("")]));
+
+        let test_cases = TestCases::default().with_range(0..2).with_indices(vec![1]);
+        check_round_trip_encoding_of_data(vec![string_array.clone()], &test_cases, HashMap::new())
+            .await;
+        let test_cases = test_cases.with_batch_size(1);
+        check_round_trip_encoding_of_data(vec![string_array], &test_cases, HashMap::new()).await;
+    }
```

Comment:
We don't actually use fixed size encoding here and we have empty string encodings elsewhere.  I don't think this test is needed.